Test/SqlDom/BackupRestoreRegressionTests.cs

```diff
@@ -0,0 +1,156 @@
+            {
+                string script = @"RESTORE DATABASE AdventureWorks2012 FROM DATABASE_SNAPSHOT = 'AdventureWorks_dbss1800'";
+                using (var scriptReader = new StringReader(script))
+                {
+                    var fragment = parser.Parse(scriptReader, out IList<ParseError> errors) as TSqlScript;
+                    if (parser is TSql80Parser)
+                    {
+                        // for the older parsers this syntax is expected to return errors
+                        Assert.AreEqual(1, errors.Count);
+                        Assert.AreEqual("Incorrect syntax near DATABASE_SNAPSHOT.", errors[0].Message);
+                    }
+                    else
+                    {
+                        // for these parser versions we expect success
+                        Assert.AreEqual(0, errors.Count);
+                        Assert.IsTrue(fragment is TSqlScript);
+                        Assert.IsTrue(fragment.Batches[0].Statements[0] is RestoreStatement);
+                    }
+                }
+            }, true);
+        }
+
+        /// <summary>
+        /// Test for BACKUP / RESTORE ... VIRTUAL_DEVICE ... syntax. This is supported in SQL Server 2005+, so parser version 90+.
+        /// </summary>
+        [TestMethod]
+        [Priority(0)]
+        [SqlStudioTestCategory(Category.UnitTest)]
+        public void BackupRestoreVirtualDevice()
+        {
+            ParserTestUtils.ExecuteTestForAllParsers(parser =>
+            {
+                string script = @"BACKUP DATABASE MyDB TO VIRTUAL_DEVICE='{A9FD7855-A978-4931-8027-F0EFB1D4F4EA}' WITH STATS = 10;
+RESTORE DATABASE MyDB FROM VIRTUAL_DEVICE='{A9FD7855-A978-4931-8027-F0EFB1D4F4EA}' WITH NORECOVERY, CHECKSUM, REPLACE, BUFFERCOUNT=16, MAXTRANSFERSIZE=2097152";
+                using (var scriptReader = new StringReader(script))
+                {
+                    var fragment = parser.Parse(scriptReader, out IList<ParseError> errors) as TSqlScript;
+                    if (parser is TSql80Parser)
+                    {
+                        // for the older parsers this syntax is expected to return errors
+                        Assert.AreEqual(2, errors.Count);
+                        Assert.AreEqual("Incorrect syntax near VIRTUAL_DEVICE.", errors[0].Message);
+                        Assert.AreEqual("Incorrect syntax near VIRTUAL_DEVICE.", errors[1].Message);
+                    }
+                    else
+                    {
+                        // for these parser versions we expect success
+                        Assert.AreEqual(0, errors.Count);
+                        Assert.IsTrue(fragment is TSqlScript);
+                        Assert.IsTrue(fragment.Batches[0].Statements[0] is BackupStatement);
+                        Assert.IsTrue(fragment.Batches[0].Statements[1] is RestoreStatement);
+                    }
+                }
+            }, true);
+        }
+	}
```

Comment:
nit: indentation

Reply:
Thanks! - addressed in 2926e5154dc4f8147178f2660b3cf2cd7b99095b.